STIPS/STIP-009.md

```diff
@@ -0,0 +1,520 @@
+
+**Reviewer**:
+
+Reviewer: []
+
+## Specification
+
+### ManagerFactory
+
+#### Structs
+
+##### InitializeParams
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+|bool|isPending|Bool if manager in pending state|  
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|factory|Address of Set Token factory|
+|mapping(address => InitializeParams)|initialize|Mapping from Set Token to initialization parameters|
+
+
+#### Functions
+
+| Name  | Caller  | Description 	|
+|------	|------	|-------------	|
+|create|deployer|Create new Set Token with a DelegatedManager manager|
+|migrate|deployer|Migrate existing Set Token to a DelegatedManager manager|
+|initialize|deployer|Initialize modules and extensions|
+
+> create
+
+```solidity
+function create(
+    address[] memory _components,
+    uint256 _units,
+    string memory _name,
+    string memory _symbol,
+    address _owner,
+    address _methodologist,
+    address[] memory _modules,
+    address[] memory _operators,
+    address[] memory _assets,
+    address[] memory _extensions
+)
+    external
+{
+    address setTokenAddress = _deploySet(
+        _components,
+        _modules,
+        _units,
+        _name,
+        _symbol
+    );
+
+    address managerAddress = _deployManager(
+        setTokenAddress,
+        _methodologist,
+        _operators,
+        _assets,
+        _extensions
+    );
+
+    initialize[setTokenAddress] = InitializeParams({
+        deployer: msg.sender,
+        owner: _owner,
+        manager: managerAddress,
+        isPending: true
+    });
+}
+```
+
+> migrate
+
+```solidity
+function migrate(
+    address memory _setTokenAddress,
+    address _owner,
+    address _methodologist,
+    address[] memory _modules,
+    address[] memory _operators,
+    address[] memory _assets,
+    address[] memory _extensions
+)
+    external
+{
+    address managerAddress = _deployManager(
+        setTokenAddress,
+        _methodologist,
+        _operators,
+        _assets,
+        _extensions
+    );
+
+    initialize[setTokenAddress] = InitializeParams({
+        deployer: msg.sender,
+        owner: _owner,
+        manager: managerAddress,
+        isPending: true
+    });
+}
+```
+
+> initialize
+
+```solidity
+function initialize(
+    address memory _setTokenAddress,
+    address[] memory _extensions,
+    mapping(address => bytes) _extensionParams
+)
+    external
+{
+    require(msg.sender == initialize[_setTokenAddress].deployer);
+    require(initialize[_setTokenAddress].isPending);
+
+    for (uint256 i = 0; i < _extensions.length; i++) {
+        _extensions[i].initialize(_extensionParams[_extensions[i]]);
```

Comment:
this may require us to use the `Address` library (which we use in `interactManager`) so basically we'd have the full initialize call encoded and passed in. Unless we can think of a better way.